lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php

```diff
@@ -344,25 +345,33 @@ public function getListTableIndexesSQL(string $table, ?string $currentDatabase =
                  WHERE oid IN (
                     SELECT indexrelid
                     FROM pg_index si, pg_class sc, pg_namespace sn
-                    WHERE ' . $this->getTableWhereClause($table, 'sc', 'sn') . ' AND sc.oid=si.indrelid AND sc.relnamespace = sn.oid
+                    WHERE ' . $this->getTableWhereClause($tableWithSchema, 'sc', 'sn') . ' AND sc.oid=si.indrelid AND sc.relnamespace = sn.oid
                  ) AND pg_index.indexrelid = oid';
     }
 
-    private function getTableWhereClause(string $table, string $classAlias = 'c', string $namespaceAlias = 'n') : string
+    /**
+     * @param string $tableWithSchema The name of the table prefixed with the schema (i.e. 'some_schema.some_table').
+     */
+    private function getTableWhereClause(string $tableWithSchema, string $classAlias = 'c', string $namespaceAlias = 'n') : string
     {
-        $whereClause = $namespaceAlias . ".nspname NOT IN ('pg_catalog', 'information_schema', 'pg_toast') AND ";
-        if (strpos($table, '.') !== false) {
-            [$schema, $table] = explode('.', $table);
+        if (strpos($tableWithSchema, '.') !== false) {
+            [$schema, $table] = explode('.', $tableWithSchema);
             $schema           = $this->quoteStringLiteral($schema);
         } else {
+            $table  = $tableWithSchema;
             $schema = "ANY(string_to_array((select replace(replace(setting,'\"\$user\"',user),' ','') from pg_catalog.pg_settings where name = 'search_path'),','))";
+            @trigger_error(sprintf(
+                'Providing a table name without a schema prefix, i.e. \'some_schema.some_table\', (%s given) is deprecated and will cause an error in Doctrine 3.0',
```

Comment:
FWIW, `master` already corresponds to `3.0.0-dev`.

Reply:
Oh, so this PR should target another branch.

Reply:
Okay? So what needs to be done?? And by who?

Reply:
This PR should target `master` and not contain any deprecations. Once this is accepted and merged, it should be rebased and filed again, this time against `2.10`. However, I still fail to see how this issue can be solved in a non-breaking way. I'd rather fix it only in `3.0` and document the proper usage for `2.10`.

Reply:
@morozov are we not using a merge up policy? IMO a non-breaking way to fix this would be to trigger a deprecation when misusing the methods, and to wrap any confusing exception in a higher level, more meaningful exception.